datalad/distribution/create_publication_target_sshwebserver.py

```diff
@@ -0,0 +1,264 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""High-level interface for creation of publication target via SSH
+"""
+
+__docformat__ = 'restructuredtext'
+
+
+import logging
+
+from os import curdir
+from os.path import join as opj, abspath, expanduser, expandvars, exists, isdir, basename, commonprefix
+
+from six.moves.urllib.parse import urlparse
+
+from datalad.support.param import Parameter
+from datalad.support.constraints import EnsureStr, EnsureNone, EnsureBool, \
+    EnsureDatasetAbsolutePath
+from datalad.support.gitrepo import GitRepo
+from datalad.support.annexrepo import AnnexRepo
+from datalad.cmd import Runner
+from datalad.cmdline.helpers import POC_get_root_handle
+from ..interface.base import Interface
+from ..interface.POC_helpers import get_submodules_dict, get_submodules_list, get_all_submodules_dict, get_git_dir, get_remotes
+from datalad.distribution.dataset import EnsureDataset, Dataset, datasetmethod, resolve_path
+from datalad.cmd import CommandError
+from datalad.utils import assure_dir, not_supported_on_windows, getpwd
+from datalad.consts import HANDLE_META_DIR, POC_STD_META_FILE
+
+
+lgr = logging.getLogger('datalad.distribution.create_publication_target_sshwebserver')
+
+
+class CreatePublicationTargetSSHWebserver(Interface):
+    """Create a target repository for publish and add it as a remote to
+    push to."""
+
+    _params_ = dict(
+        # TODO: Somehow the replacement of '_' and '-' is buggy on
+        # positional arguments
+        # TODO: Figure out, whether (and when) to use `sshurl` as push url
+        dataset=Parameter(
+            args=("--dataset", "-d",),
+            doc="""specify the dataset to create the publication target for. If
+                no dataset is given, an attempt is made to identify the dataset
+                based on the current working directory""",
+            constraints=EnsureDataset() | EnsureNone()),
+        target=Parameter(
+            args=('target',),
+            doc="""Sibling name to create for this publication target.
+                If RECURSIVE is set, the same name will be used to address
+                the subdatasets' siblings""",
+            constraints=EnsureStr() | EnsureNone()),
+        sshurl=Parameter(
+            args=("sshurl",),
+            doc="""SSH URL to use to create the target sibling(s)""",
+            constraints=EnsureStr() | EnsureNone()),
```

Comment:
May be. It's about setting up the target repos. So we need ssh to create them. Doesn't necessarily mean, we need ssh to push.